lib/matplotlib/dates.py

```diff
@@ -271,6 +280,13 @@ def _dt64_to_ordinalf(d):
     return dt
 
 
+def _dt64_to_ordinalf_iterable(d):
+    dnew = np.zeros(len(d))
+    for nn, dd in enumerate(d):
+        dnew[nn] = _dt64_to_ordinalf(dd)
+    return(dnew)
```

Comment:
Does this work as well:
~~~
return np.array(_dt64_to_ordinalf(dd) for dd in d)
~~~

Reply:
That fails some tests and returns `<generator object _dt64_to_ordinalf_iterable.<locals>.<genexpr>` :-(

Reply:
Ok, maybe this is numpy-version dependent. Anyway, there has to be a way to not add each element one-by-one on the python level, which is slow. My next guess would be:

~~~
return np.from_iter((_dt64_to_ordinalf(dd) for dd in d), float, count=len(d))
~~~

and if all fails, we can still create a list-comprehension and turn that into an array:

~~~
return np.array([_dt64_to_ordinalf(dd) for dd in d])

~~~

Reply:
`np.fromiter` works.  Not at all sure I understand why...